src/impy/util.py

```diff
@@ -214,6 +221,88 @@ def readOutput(self):
             self.capturedtext += char
 
 
+# Functions to check and download dababase files on github
+
+
+def _download_file(outfile, url):
+    """Download a file from 'url' to 'outfile'"""
+    fname = pathlib.Path(url).name
+    try:
+        response = urllib.request.urlopen(url)
+    except BaseException:
+        raise ConnectionError(
+            f"_download_file: probably something wrong with url = '{url}'"
+        )
+    total_size = response.getheader("content-length")
+
+    min_blocksize = 4096
+    if total_size:
+        total_size = int(total_size)
+        blocksize = max(min_blocksize, total_size // 100)
+    else:
+        blocksize = min_blocksize
+
+    wrote = 0
+    with open(outfile, "wb") as f:
+        while True:
+            data = response.read(blocksize)
+            if not data:
+                break
+            f.write(data)
+            wrote += len(data)
+            if total_size:
+                print(
+                    f"Downloading {fname}: {wrote/total_size*100:.0f}% "
+                    f"done ({wrote/(1024*1024):.0f} Mb) \r",
+                    end=" ",
+                )
+            else:
+                print(
+                    f"Downloading {fname}: {wrote/(1024*1024):.0f} Mb downloaded \r",
+                    end=" ",
+                )
+    print()
```

Comment:
It looks like you took my advice and don't depend on tqdm anymore, so can we remove it from the install requirements?

Reply:
tqdm is removed. Now the progress is shown as a percentage of loaded. I will check your approach.